moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -66,312 +61,279 @@ Eigen::Isometry3d createTestIsometry()
-  // configurations.
-  random_numbers::RandomNumberGenerator rng(0);
-
   KDL::Tree kdl_tree;
   if (!kdl_parser::treeFromUrdfModel(*robot_model->getURDF(), kdl_tree))
   {
     st.SkipWithError("Can't create KDL tree.");
     return;
   }
 
-  KDL::Chain kdl_chain;
-  if (!kdl_tree.getChain(jmg->getJointModels().front()->getParentLinkModel()->getName(),
-                         jmg->getLinkModelNames().back(), kdl_chain))
-  {
-    st.SkipWithError("Can't create KDL Chain.");
-    return;
-  }
+  KDL::TreeJntToJacSolver jacobian_solver(kdl_tree);
+  KDL::Jacobian jacobian(kdl_tree.getNrOfJoints());
+  KDL::JntArray kdl_q(kdl_tree.getNrOfJoints());
+  const std::string tip_link = jmg->getLinkModelNames().back();
 
-  KDL::ChainJntToJacSolver jacobian_solver(kdl_chain);
+  // Manually seeded RandomNumberGenerator for deterministic results
+  random_numbers::RandomNumberGenerator rng(0);
 
   for (auto _ : st)
   {
     // Time only the jacobian computation, not the forward kinematics.
     st.PauseTiming();
-    kinematic_state.setToRandomPositions(jmg, rng);
-    kinematic_state.updateLinkTransforms();
-    KDL::Jacobian jacobian(kdl_chain.getNrOfJoints());
-    KDL::JntArray kdl_q;
-    kdl_q.resize(kdl_chain.getNrOfJoints());
-    kinematic_state.copyJointGroupPositions(jmg, &kdl_q.data[0]);
+    state.setToRandomPositions(jmg, rng);
+    state.copyJointGroupPositions(jmg, &kdl_q.data[0]);
     st.ResumeTiming();
-    jacobian_solver.JntToJac(kdl_q, jacobian);
+    jacobian_solver.JntToJac(kdl_q, jacobian, tip_link);
   }
 }
 
-BENCHMARK(multiplyAffineTimesMatrix)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(multiplyMatrixTimesMatrix)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(multiplyIsometryTimesIsometry)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-
-BENCHMARK(inverseIsometry3d)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseAffineIsometry)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseAffine)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseMatrix4d)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-
-BENCHMARK(robotStateConstruct)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateCopy)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateUpdate)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateForwardKinematics)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
-
-BENCHMARK(moveItJacobian);
-BENCHMARK(kdlJacobian);
+BENCHMARK(multiplyAffineTimesMatrixNoAlias);
+BENCHMARK(multiplyMatrixTimesMatrixNoAlias);
+BENCHMARK(multiplyIsometryTimesIsometryNoAlias);
+BENCHMARK(multiplyMatrixTimesMatrix);
+BENCHMARK(multiplyIsometryTimesIsometry);
```

Comment:
I just tested different variants. It's quite obvious that NoAlias versions are faster.

Reply:
Good to know about the `.noalias()`. Unfortunately the default behavior is much more frequent in code, so it would be great to keep both versions of the three multiply benchmarks.